azurerm/internal/services/vmware/vmware_cluster_resource.go

```diff
@@ -0,0 +1,210 @@
+package vmware
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/avs/mgmt/2020-03-20/avs"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceVmwareCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVmwareClusterCreate,
+		Read:   resourceVmwareClusterRead,
+		Update: resourceVmwareClusterUpdate,
+		Delete: resourceVmwareClusterDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Hour),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Hour),
+			Delete: schema.DefaultTimeout(5 * time.Hour),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ClusterID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"private_cloud_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.PrivateCloudID,
+			},
+
+			"cluster_size": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(3, 16),
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"av20",
+					"av36",
+					"av36t",
+				}, false),
+			},
+
+			"cluster_id": {
```

Comment:
how is this different from the resource.id?

Reply:
It will have a self-defined cluster_id, like "1","2","3"... Shall we call it "cluster_name"?

Reply:
I've changed to `cluster_name`

Reply:
what does this 1,2,3 represent? cluster name is resource.name, cluster_number would differentiate it from both id and name

Reply:
For example, it represents a private cloud has three clusters. And the cluster_name will be 1, 2,3, id in this private cloud